### PR TITLE
Cargo Router Door Timing Fixes

### PR DESCRIPTION
**Pull request by Nexusuxen**
[BUG]

## About the PR 

Fixes the issues detailed in #6607 which resulted from PR #6533

Changes the AMDL to use a variable instead of a hardcoded number for determining how long to keep cargo doors open for and for how long it should wait before trying to load something else. This is defaulted to 3 seconds, and most/all loaders who had doors close prematurely have been adjusted accordingly on: Cogmap1, Cogmap2, Kondaru, and Ozymandius. There may still need to be a couple tweaks here and there, as I was initially over-generous with how long to set the delays, and may have missed one or two when sifting through and correcting the overly long delays.

## Why's this needed? 

another bug. i squash, again. ((also lets mappers adjust how long they want to keep AMDL/mass driver doors open for))

## Changelog

```changelog
(u)Nexusuxen
(*)Cargo router doors should now stay open for long enough to allow for cargo to not get stuck in space.
```

### CODE REVIEW

File: code/obj/machinery/launcherloader.dm [hunk 1]
```diff
@@ -19,6 +19,8 @@
 	var/operating = 0
 	var/driver_operating = 0
 	var/trash = 0
+	/// Amount of time *in seconds* before connected blast doors should close
+	var/door_delay = 3
```
Comment (ZeWaka) [inline code/obj/machinery/launcherloader.dm:23]: if it's in seconds then say so here
```suggestion
	var/door_delay = 3 SECONDS
```

File: code/obj/machinery/launcherloader.dm [hunk 2]
```diff
@@ -64,11 +66,11 @@
 						SPAWN_DBG(0)
 							if (door)
 								door.open()
-						SPAWN_DBG(3 SECONDS)
+						SPAWN_DBG(door_delay SECONDS)
```
Comment (ZeWaka) [inline code/obj/machinery/launcherloader.dm:69]:
```suggestion
						SPAWN_DBG(door_delay)
```

File: code/obj/machinery/launcherloader.dm [hunk 3]
```diff
@@ -64,11 +66,11 @@
...
 							if (door)
 								door.close()
 
-				SPAWN_DBG(door ? 30 : 20) driver_operating = FALSE
+				SPAWN_DBG(door ? door_delay SECONDS : 20) driver_operating = FALSE
```
Comment (ZeWaka) [inline code/obj/machinery/launcherloader.dm:73]:
```suggestion
				SPAWN_DBG(door ? door_delay : 2 SECONDS) driver_operating = FALSE
```

File: code/obj/machinery/launcherloader.dm [hunk 4]
```diff
@@ -19,6 +19,8 @@
 	var/operating = 0
 	var/driver_operating = 0
 	var/trash = 0
+	/// Amount of time *in seconds* before connected blast doors should close
```
Comment (ZeWaka) [inline code/obj/machinery/launcherloader.dm:22]:
```suggestion
	/// Amount of time before connected blast doors should close
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/launcherloader.dm (hunk 4 above)
Comment (Kubius) [reply code/obj/machinery/launcherloader.dm]: It's multiplied by seconds, why is it bad to specify seconds

**Comment by Kubius** [comment]
Double checking with waka approach: if defines can be used in varedits, this change is fine, but that does have to be the case (don't know if it is).

**Comment by Nexusuxen** [comment]
Can't figure out how to reply to a comment on mobile oop

Regardless: That is a very good point. You could still var edit and slap a 0 on the end of your number to turn it from seconds to deciseconds, but that's unintuitive and would be weird to put into documentation

**Comment by ZeWaka** [comment]
> Double checking with waka approach: if defines can be used in varedits, this change is fine, but that does have to be the case (don't know if it is).

Defines cannot be used in varedits. I don't see any issues for this as a result though.

**Comment by Nexusuxen** [comment]
I don't directly see the merit of having the SECONDS being in the variable declaration. But, I can tweak it so that it multiplies by SECONDS in New() rather than the variable declaration itself (so you don't multiply SECONDS by SECONDS).

**Comment by Nexusuxen** [comment]
I presume the merit of having the simplified door_delay was to reduce clutter where it was called by removing the SECONDS portion, so I went with a compromise that wouldn't confuse mappers (as this is var primarily only really useful when you're mapping) but also would add to legibility.

Nothing broke during tests too, which is a plus.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**